wallet-android/src/main/java/com/iohk/jormungandrwallet/Wallet.kt

```
@@ -0,0 +1,30 @@
+package com.iohk.jormungandrwallet
+
+class Wallet {
+    private var walletPtr: Long = 0
+
+
+    private external fun recover(mnemonics: String) : Long
+    private external fun delete(wallet: Long)
+    private external fun totalValue(wallet: Long) : Int
+    private external fun initialFunds(wallet: Long, block0: ByteArray) : Long
+    companion object {
+        init {
+            System.loadLibrary("wallet_jni")
+        }
+    }
+
+    constructor(mnemonics: String) {
+        this.walletPtr = recover(mnemonics)
+        assert(this.walletPtr != 0.toLong())
+    }
```

can we throw an error instead here? and document the function may fail